tests/iothreads_test.go

```diff
@@ -0,0 +1,206 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"os/exec"
+	"time"
+	"encoding/xml"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("IOThreads", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	gracePeriod := int64(30)
+	sharedPolicy := v1.IOThreadsPolicyShared
+	_true := true
+	IOThreadSpec := v1.VirtualMachineInstanceSpec{
+	    TerminationGracePeriodSeconds: &gracePeriod,
+	    Domain: v1.DomainSpec{
+		Resources: v1.ResourceRequirements{
+		    Requests: k8sv1.ResourceList{
+			k8sv1.ResourceMemory: resource.MustParse("1024M"),
+			k8sv1.ResourceCPU: resource.MustParse("2"),
+		    },
+		},
+		IOThreadsPolicy: &sharedPolicy,
+
+		Devices: v1.Devices{
+		    Disks: []v1.Disk{
+			{
+			    Name:       "ded1",
+			    VolumeName: "ded1volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			    DedicatedIOThread: &_true,
+			},
+			{
+			    Name:       "shr1",
+			    VolumeName: "shr1volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			},
+			{
+			    Name:       "shr2",
+			    VolumeName: "shr2volume",
+			    DiskDevice: v1.DiskDevice{
+				Disk: &v1.DiskTarget{
+				    Bus: "virtio",
+				},
+			    },
+			},
+		    },
+		},
+	    },
+	    Volumes: []v1.Volume{
+		{
+		    Name: "ded1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr1volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+		{
+		    Name: "shr2volume",
+		    VolumeSource: v1.VolumeSource{
+			EmptyDisk: &v1.EmptyDiskSource{
+			    Capacity: resource.MustParse("1G"),
+			},
+		    },
+		},
+	    },
+	}
+
+	var IOThreadVMI *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+		IOThreadVMI = tests.NewRandomVMI()
+		IOThreadVMI.Spec = IOThreadSpec
+	})
+
+	Context("IOThreads Policies", func() {
+
+		// CNV-1046
+		It("Virtual Disk Settings - IOThreads", func() {
+
+			// How many symbols in the names of pod and VMI shoul match
+			symbols_to_compare := 30
+
+			IOThreadVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(IOThreadVMI)
+			// Step 1
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(IOThreadVMI.Name) > symbols_to_compare).To(BeTrue())
```

Comment:
There is better way of using the Expect. You can place annotation in the Expect. When it fails, the label is than shown and produce much clearer information of failure. For instance you code here will output:
```
Expected false to be true.
```
But when using the label `Expect(len(IOThreadVMI.Name) > symbols_to_compare).To(BeTrue(), "Should have some symbols...") will output
```
Should have some symbols...:
Expected false to be true.
```
which is much cleaner to read.

Please update the expects.

Reply:
I've added comments to all Expect statements except "Expect(err).ToNot(HaveOccurred())"